clippy_lints/src/trait_bounds.rs

```diff
@@ -167,6 +167,52 @@ impl<'tcx> LateLintPass<'tcx> for TraitBounds {
             }
         }
     }
+
+    fn check_ty(&mut self, cx: &LateContext<'tcx>, ty: &'tcx Ty<'tcx>) {
+        if_chain! {
+            if let TyKind::Ref(.., mut_ty) = &ty.kind;
+            if let TyKind::TraitObject(bounds, ..) = mut_ty.ty.kind;
+            if bounds.len() > 2;
+            then {
+                let mut bounds_span = bounds[0].span;
+
+                for bound in bounds.iter().skip(1) {
+                    bounds_span = bounds_span.to(bound.span);
+                }
+
+                let mut seen_def_ids = FxHashSet::default();
+                let mut fixed_traits = Vec::new();
+
+                for bound in bounds.iter() {
```

Comment:
nit: comment explaining why this works

Reply:
I still think this needs explanation: what is "fixed" here? At what stage is this code iterating over supertrait bounds? It's not quite clear.

Reply:
(Sorry about focusing on this, but while this works and I think I have an idea as to why it works; it's not immediately clear and we should try and be clear here)

Reply:
No worries, always happy to make my code clearer.

Have written up some more that should hopefully clear it up